cmd/sops/common/common.go

```diff
@@ -136,3 +140,178 @@ func DefaultStoreForPath(path string) Store {
 	}
 	return &json.BinaryStore{}
 }
+
+const KMS_ENC_CTX_BUG_FIXED_VERSION = "3.3.0"
+
+func DetectKMSEncryptionContextBug(tree *sops.Tree) (bool, error) {
+	versionCheck, err := version.AIsNewerThanB(KMS_ENC_CTX_BUG_FIXED_VERSION, tree.Metadata.Version)
+	if err != nil {
+		return false, err
+	}
+
+	if versionCheck {
+		_, _, key := GetKMSKeyWithEncryptionCtx(tree)
+		if key != nil {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func GetKMSKeyWithEncryptionCtx(tree *sops.Tree) (kgndx int, keyndx int, key *kms.MasterKey) {
+	for i, kg := range tree.Metadata.KeyGroups {
+		for n, k := range kg {
+			kmsKey, ok := k.(*kms.MasterKey)
+			if ok {
+				if kmsKey.EncryptionContext != nil && len(kmsKey.EncryptionContext) >= 2 {
```

Comment:
It's unlikely that it'll happen, but this could have false positives. The bug is only triggered if there's at least two unique values for the encryption context